lib/fine-types/app/Commands/Log.hs

```diff
@@ -0,0 +1,14 @@
+module Commands.Log where
+
+import Prelude
+
+import Options.Log (LogFile (..))
+import System.IO (Handle, IOMode (..), hPutStrLn, stderr, stdout, withFile)
+
+withLogHandle :: LogFile -> (Handle -> IO a) -> IO a
+withLogHandle StdOut f = f stdout
+withLogHandle StdErr f = f stderr
+withLogHandle (LogFile fp) f = withFile fp WriteMode f
+
+withLogPutLn :: LogFile -> ((String -> IO ()) -> IO a) -> IO a
```

Comment:
```suggestion
withLogTracer :: LogFile -> (Tracer IO String -> IO a) -> IO a
```

This is beginning to look like a `Control.Tracer` that we also have in the wallet. 🤓 Not necessarily in scope for this pull request, but in some future, using [contra-tracer][] may be a good option for logging — it's very simple, yet effective.

 [contra-tracer]: https://hackage.haskell.org/package/contra-tracer-0.2.0.0/docs/Control-Tracer.html